erpnext/crm/doctype/contract/contract.py

```diff
@@ -67,7 +153,41 @@ def update_fulfilment_status(self):
 
 	def get_fulfilment_progress(self):
 		return len([term for term in self.fulfilment_terms if term.fulfilled])
+		
+
+@frappe.whitelist()
+def accept_contract_terms(sign, signee, contract=None):
+	"""
+	Gets signee: whoever signed the contract, contract: contract name
 
+	updates is_signed, signee and signed_on field in db
+
+	returns relevant message
+	"""
+	try:
+		doc = frappe.get_doc("Contract", contract)
+		doc.is_signed = 1
+		doc.signature = sign
+		doc.signee = signee
+		doc.signed_on = now_datetime()
+		doc.save()
+		return "Your contract is shared successfully!"
+	except:
+		return "couldn't submit signing"
```

Comment:
Returning this won't do anything. If you want to throw an exception, you'll have to do `frappe.throw("Exception message")`